daft/functions/misc.py

```diff
@@ -725,3 +726,141 @@ def slice(expr: Expression, start: int | Expression, end: int | Expression | Non
         (Showing first 3 of 3 rows)
     """
     return Expression._call_builtin_scalar_fn("slice", expr, start, end=end)
+
+
+def when(condition: Expression | bool, value: Expression | Any) -> WhenExpr:
```

Comment:
```suggestion
def when(condition: Expression | bool, then: Expression | Any) -> WhenExpr:
```

i think it's a bit nicer if it's named `then` as then I can use it as a kwarg (if i want to), which is IMO a bit more readable. 

```py
when(df["x"] == 3, then="three")
.when(df["x"] == 4, then="four")
```

Reply:
Good idea, I'll do that